sktime/forecasting/theta.py

```diff
@@ -0,0 +1,218 @@
+            )
+
+        super(ThetaForecaster, self).__init__(smoothing_level=smoothing_level)
+
+    def _to_nested(self, y):
+        nested = pd.DataFrame(pd.Series([y]))
+
+        return nested
+
+    def fit(self, y, fh=None):
+        """
+        Fit to training data.
+
+        Parameters
+        ----------
+
+        y : pandas.Series
+            Target time series to which to fit the forecaster.
+        fh : array-like, optional (default=[1])
+            The forecasters horizon with the steps ahead to to predict.
+
+        Returns
+        -------
+
+        self : returns an instance of self.
+        """
+
+        y = validate_y(y)
+        self._set_fh(fh)
+
+        # update observation horizon
+        self._set_obs_horizon(y.index)
+
+        if self._deseasonaliser.sp == 1:
+            self._is_seasonal = False
+        else:
+            self._is_seasonal = seasonality_test(y, freq=self._deseasonaliser.sp)
+
+        if self._is_seasonal:
+            y_nested = self._to_nested(y)
+            y = self._deseasonaliser.fit_transform(y_nested).iloc[0, 0]
+
+        # Find theta lines.
+
+        # Theta lines are just SES + drift.
+        super().fit(y, fh=fh)
+        self.smoothing_level_ = self._fitted_estimator.params["smoothing_level"]
+
+        # Drift calculated through least squares regression.
+        coefs = fit_trend(y.values.reshape(1, -1), order=1)
+        self.trend_ = coefs[0, 0] / 2
+
+        self._is_fitted = True
+
+        return self
+
+    def predict(self, fh=None, X=None, return_conf_int=False, alpha=ExpSmoothingForecaster._DEFAULT_ALPHA):
+        """
+        Make forecasts.
+
+        Parameters
+        ----------
+
+        fh : array-like
+            The forecasters horizon with the steps ahead to to predict. Default is
+            one-step ahead forecast, i.e. np.array([1]).
+
+        Returns
+        -------
+
+        y_pred : pandas.Series
+            Returns series of predicted values.
+        """
+
+        self._check_is_fitted()
+
+        # Set forecast horizon.
+        self._set_fh(fh)
+
+        # SES
+        y_pred = super().predict(fh=fh)
+
+        if np.isclose(self.smoothing_level_, 0.0):
+            # SES was constant so revert to simple trend
+            drift = self.trend_ * fh
+        else:
+            # Calculate drift from SES parameters
+            n_obs = len(self._obs_horizon)
+            drift = self.trend_ * (
+                fh + (1 - (1 - self.smoothing_level_) ** n_obs) / self.smoothing_level_
+            )
+
+        y_pred += drift
+
+        if self._is_seasonal:
+            # Reseasonalise.
+            y_pred_nested = self._to_nested(y_pred)
+            y_pred = self._deseasonaliser.inverse_transform(y_pred_nested).iloc[0, 0]
+
+        if return_conf_int:
+            errs = self.pred_errs(alpha=alpha)
+            return y_pred, errs
+
+        return y_pred
+
+    def pred_errs(self, alpha=ExpSmoothingForecaster._DEFAULT_ALPHA):
```

Comment:
Would keep consistent naming, so something like `compute_conf_int()`. Should this be hidden, so that users should mainly work through the `predict` method?

Reply:
The reason it's also available as a separate method is so that you have the ability to compute intervals separately. This could be useful if e.g. your forecaster is part of a pipeline and changing the output would break the pipeline, but you might want to go back to the forecaster later to get the uncertainties.

Please see my point in the review of #218 though - I'm not sure it's correct to talk about "confidence" intervals for a prediction. Also the current approach computes error margins, not intervals. Can you advise on what naming convention you'd like to go with before I make the changes?